### PR TITLE
Np 46040 nvi admin search

### PR DESCRIPTION
**Pull request by truhacevkir**
Adding support to be able to search for all nvi candidates when nvi-admin.

Main changes:

Removing "mandatory default" customer filter when searching nvi candidates without providing affilition/customer.
Removing "mandatory default" year filter when searching nvi candidates without providing year.

This makes it possible for nvi-admin list nvi-candidates and get same "view" as nvi-curators at other institutions, in addition to list all nvi candidates.

### CODE REVIEW

File: index-handlers/src/main/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandler.java [hunk 1]
```diff
@@ -129,9 +143,12 @@ private void assertUserIsAllowedToSearchAffiliations(List<URI> affiliations, URI
...
 
-    private String extractQueryParamPublicationDateOrDefault(RequestInfo requestInfo) {
-        return requestInfo.getQueryParameters()
-            .getOrDefault(QUERY_PARAM_YEAR, String.valueOf(ZonedDateTime.now().getYear()));
+    private String extractQueryParamPublicationDateOrDefault(RequestInfo requestInfo, boolean isAdmin) {
+        var yearQueryParam =  requestInfo.getQueryParameters()
+            .getOrDefault(QUERY_PARAM_YEAR, null);
+        return isAdmin
+                   ? yearQueryParam
+                   : nonNull(yearQueryParam) ? yearQueryParam : String.valueOf(ZonedDateTime.now().getYear());
```
Comment (truhacevkir) [inline index-handlers/src/main/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandler.java:151]: admin can list all candidates not depending on year

File: event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinMapper.java [hunk 2]
```diff
@@ -94,12 +93,16 @@ private static ZonedDateTime getZonedDateTime(CristinNviReport cristinNviReport)
...
     private static URI constructPublicationBucketUri(String publicationIdentifier) {
-        return UriWrapper.fromUri(PERSISTED_RESOURCES_BUCKET)
+        return UriWrapper.fromHost(PERSISTED_RESOURCES_BUCKET)
                    .addChild("resources")
-                   .addChild(publicationIdentifier)
+                   .addChild(publicationIdentifierKeyInS3Bucket(publicationIdentifier))
                    .getUri();
     }
 
+    private static String publicationIdentifierKeyInS3Bucket(String publicationIdentifier) {
```
Comment (sondrev) [inline event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinMapper.java:102]: or just "withGzExtension"?

File: index-handlers/src/main/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandler.java (hunk 1 above)
Comment (sondrev) [reply index-handlers/src/main/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandler.java:151]: nested elvis (?:) are hard to read.
Will this do the same?

(isAdmin || nonNull(yearQueryParam))
  ? yearQueryParam
  : String.valueOf(ZonedDateTime.now().getYear())

Comment (sondrev) [reply index-handlers/src/main/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandler.java:151]: Also I feel like you are giving this method too much responibility compared with what the other "ExtractQueryParam**OrDefault" methods does. Could be better to just return the first value, and let the parent caller do the admin-check and decide which year to pass on

**Review by sondrev — APPROVED** [review]

**Pull request merged**